### ISSUE TITLE
Add support for only/skip to modules

### ISSUE THREAD

**Issue by Arkni** [qunitjs/qunit#1163]
These features are similar to the already implemented ones of tests. Usually, when I'm debugging a large codebase, I often only want to run a subset of tests (in my case it's a `module`). The same goes when I want to skip a chunk of tests.

About the syntax, we can either use:
```js
const { module, test } = QUnit;

// Normal
module("Module A", () => {
  test("test A", assert => {
    assert.expect(1);
    assert.ok( true );
  });
});

// Only
module.only("Module B", () => {
  test("test B", assert => {
    assert.expect(1);
    assert.ok(true);
  });
});

// Skip
module.skip("Module C", () => {
  test("test C", assert => {
    assert.expect( 1 );
    assert.ok( true );
  });
});
```

Or, we can use the second parameter of `QUnit.module()` to define whether a module should be the only one to run or the one that should be skipped:
```js
module("Module A", { only: true }, () => {
  // ...
});

module("Module B", { skip: true }, () => {
  // ...
});
```

The second approach may change the semantic of the second parameter as it's used to define hooks.

I might be able to implement something for the next weekend in case these features benefit someone.

**Comment by trentmwillis** [comment]
I agree that this is definitely a feature we should support.

That said, I don't particularly like the second approach due to changing semantics as you've pointed out, but the first approach also doesn't feel quite right. For individual tests, the functions are:

```js
QUnit.test();
QUnit.skip();
QUnit.todo();
QUnit.only();
```

But for module's were proposing:

```js
QUnit.module();
QUnit.module.skip();
QUnit.module.todo(); // Unsure if this is needed, but would be nice for completeness
QUnit.module.only();
```

I think if we go this route, we likely will also want to support:

```js
QUnit.test();
QUnit.test.skip();
QUnit.test.todo(); // Unsure if this is needed, but would be nice for completeness
QUnit.test.only();
```

So that it is consistent. This is also very similar to how Mocha supports these features. Thoughts?

**Comment by Arkni** [comment]
> I don't particularly like the second approach

I don't like it either. I just wanted to point that there is an other approach that we can follow.

But the way, I was going to propose implementing the first approach for `test` too, but changed my mind as you already have different methods that do the same and I was thinking that you may see the idea as bad and don't want to create new aliases.

> I think if we go this route, we likely will also want to support

Yes, for consistency, we may end up implementing the same (Of course only in case these features are welcome).

> This is also very similar to how Mocha supports these features

Exactly. `Jest` do the same too.

**Comment by Arkni** [comment]
@trentmwillis 

What should be the proper way to handle `QUnit.module.todo()`  ?
Should it mark its tests including the tests of the child suites as todos ?

**Comment by trentmwillis** [comment]
@Arkni yeah, that's what I would expect the behavior to be. We should be sure to explicitly define how `skip` or `todo` tests within a module that is `todo` or `skip` work.

**Comment by Arkni** [comment]
I pushed my first implementation of these features and created #1168 
Waiting for your feedback :)

**Comment by trentmwillis** [comment]
`QUnit.module.only` has been added via: https://github.com/qunitjs/qunit/pull/1179.

Still need `skip` and `todo`.

**Comment by Arkni** [comment]
`QUnit.module.skip` and `QUnit.module.todo` have been added respectively via: #1193 & #1195

**Comment by trentmwillis** [comment]
Documentation for this has now landed as well! These features will be available in the next release (likely `2.4.0`), which will hopefully be cut soon (got a few more things to land first).
